### DISCUSSION TITLE
TANDialog image datatypes

### DISCUSSION THREAD

**Discussion by cwonschak** [libfintx/libfintx#18] [Ideas]
Hi,

i wanted to ask if it would be possible to provide the tan images as byte data in the TANDialog, optional to the SixLabs Image format. It simply feels wrong to add an extra dependency (chargeable) to our project only to access the image data before displaying it.

Best regards,
Christian

**Comment by leo-schick** [comment]
I think that would be possible. Could you provide the function you’re referring to which returns the SixLabs Image?

**Comment by cwonschak** [comment]
Hi,

thanks for replying :)

The properties i am talking about are **MatrixImage** and **FlickerImage** of the TANDialog class, which are of the SixLabors datatype Image\<Rgba32\>. 

The data for them is for example provided by the **MatrixCode** class. It parses the TAN string and writes the extracted byte array to a memory stream. That memory stream is then loaded by the Image<Rgba32> method of the SixLabors Image class. It would be nice if that byte array itself was somehow stored in a property of the TANDialog class (e.g. MatrixImageAsByteArray) than just the converted Image\<Rgba32\> data.

Same goes for the method **RenderAsGif()** of the **FlickerRenderer** class that provides the data for the FlickerImage property of the TANDialog.

Something like that, so the image data is accessible without the SixLabors dependency.

Greetings,
Christian

**Reply by leo-schick** [reply]
I created now a PR for this. Please check https://github.com/libfintx/libfintx/pull/19

**Reply by cwonschak** [reply]
Thats great, this helps a lot. Thank you very much @leo-schick for adding this option :)

**Comment by leo-schick** [comment]
I released a new version 1.2.0 which covers this. Discussion closed.
